### ISSUE TITLE
Starred expression not checked when used in function argument

### ISSUE THREAD

**Issue by abgros** [astral-sh/ty#2582]
### Summary

This incorrect code passes type checking:

```python
def some_fn(a: int):
    pass

some_fn(*None)
```
But fails at runtime:
```
TypeError: some_fn() argument after * must be an iterable, not NoneType
```

On the other hand, this is correctly flagged as an error:
```python
b = *None
```

https://play.ty.dev/952eb26e-1acb-481f-b419-8e15f3c2a84a

### Version

ty 0.0.7

**Comment by AlexWaygood** [comment]
Thanks! Definitely a bug.

**Comment by denyszhak** [comment]
@AlexWaygood Would love to give this one a shot

**Comment by carljm** [comment]
@denyszhak sure
